File: src/android/framework/PathFigure.java
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------
package Windows.UI.Xaml.Controls;

import android.graphics.PointF;

public class PathFigure {
    boolean _isClosed;
    PointF _startPoint;
    boolean _isFilled;
    PathSegmentCollection _segments;

    public boolean getIsClosed() {
        return _isClosed;
    }
    public void setIsClosed(boolean isClosed) {
        _isClosed = isClosed;
    }

    public PointF getStartPoint() {
        return _startPoint;
    }
    public void setStartPoint(PointF startPoint) {
        _startPoint = startPoint;
    }

    public boolean getIsFilled() {
        return _isFilled;
    }
    public void setIsFilled(boolean isFilled) {
        _isFilled = isFilled;
    }

    public PathSegmentCollection getSegments() {
        return _segments;
    }
    public void setSegments(PathSegmentCollection segments) {
        _segments = segments;
    }
}

File: www/framework/PathFigure.js
//-------------------------------------------------------------------------------------------------------
// Copyright (C) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//-------------------------------------------------------------------------------------------------------

//
// A portion of a geometry.
//
function PathFigure() {
    ace.UIElement.call(this, "Windows.UI.Xaml.Media.PathFigure");
};

// Inheritance
PathFigure.prototype = Object.create(ace.NativeObject.prototype);

PathFigure.prototype.getIsClosed = function () { return this.get("PathFigure.IsClosed"); };
PathFigure.prototype.setIsClosed = function (isClosed) { this.set("PathFigure.IsClosed", isClosed); };

PathFigure.prototype.getIsFilled = function () { return this.get("PathFigure.IsFilled"); };
PathFigure.prototype.setIsFilled = function (isFilled) { this.set("PathFigure.IsFilled", isFilled); };

PathFigure.prototype.getSegments = function () { return this.get("PathFigure.Segments"); };
PathFigure.prototype.setSegments = function (segments) { this.set("PathFigure.Segments", segments); };

PathFigure.prototype.getStartPoint = function () { return this.get("PathFigure.StartPoint"); };
PathFigure.prototype.setStartPoint = function (point) { this.set("PathFigure.StartPoint", point); };

module.exports = PathFigure;
